### PR TITLE
handle all cases for picking objects

### PR DESCRIPTION
**Pull request by tixvage**

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -112,15 +112,28 @@ Item Map::use(int x, int y, Item item, bool manual) {
...
+                case Item::BOX:
+                    return get_tile_ref(x, y).remove_box_item();
+                case Item::APPLE:
+                    if(tile.is_apple_grow())
+                        get_tile_ref(x, y).set_apple_grow(0);
+                    else
+                        get_tile_ref(x, y) = Tile(Item::EMPTY);
+                    return Item::APPLE;
+
+                case Item::FIRE:
```
Comment (MFaik) [inline src/map.cpp:125]: you need to add comment explaining this line

File: src/map.cpp [hunk 2]
```diff
@@ -112,15 +112,28 @@ Item Map::use(int x, int y, Item item, bool manual) {
...
+                case Item::LAVA:
+                case Item::ROBOT:
+                case Item::BASE:
+                case Item::PATH:
+                case Item::EMPTY:
+                    break;
+                case Item::STONE:
```
Comment (MFaik) [inline src/map.cpp:133]: also a comment here

**Review by MFaik — CHANGES_REQUESTED** [review]
you can check the rest of the code to see how I comment these cases

File: src/map.cpp (hunk 2 above)
Comment (tixvage) [reply src/map.cpp:133]: okey

**Pull request merged**